metrics-util/src/registry/recency.rs

```diff
@@ -217,6 +217,7 @@ impl GenerationalAtomicStorage {
 /// tracking recency does not matter, despite their otherwise tight coupling.
 pub struct Recency<K> {
     mask: MetricKindMask,
+    #[allow(clippy::type_complexity)]
```

Comment:
The same recommendation to extract a type alias here. This code was easier to understand, but I couldn't think of a good name for the extracted type(s)

Reply:
I'm going to leave this one as just being an allow